ratchet_core/src/ws.rs

```diff
@@ -191,25 +190,9 @@ where
                         trace!("Received an unsolicited pong frame");
                         Ok(Message::Pong(payload.freeze()))
                     } else {
-                        if control_buffer[..].eq(&payload[..]) {
-                            control_buffer.clear();
-                            trace!("Received pong frame");
-                            Ok(Message::Pong(payload.freeze()))
-                        } else {
-                            trace!("Received a pong frame with an incorrect payload. Closing the connection");
-                            self.closed = true;
-                            self.framed
-                                .write_close(CloseReason {
-                                    code: CloseCode::Protocol,
-                                    description: Some(CONTROL_DATA_MISMATCH.to_string()),
-                                })
-                                .await?;
-
-                            return Err(Error::with_cause(
-                                ErrorKind::Protocol,
-                                CONTROL_DATA_MISMATCH.to_string(),
```

Comment:
So does this mean the bodies are essentially irrelevant if it's not required to check them?

Reply:
The spec states that the response body to a Ping must have the exact same payload. However, it never states that the connection should be failed if it is incorrect and a peer may send multiple Ping frames, and the recipient _may_ elect to only respond to the most recent frame.